session.go

```diff
@@ -718,6 +793,51 @@ func (s *session) handleUnpackedPacket(packet *unpackedPacket, rcvTime time.Time
 
 	r := bytes.NewReader(packet.data)
 	var isAckEliciting bool
+	var recoverable bool
+
+	if s.mitigationOn(packet.encryptionLevel) {
+		var ignorableError *qerr.QuicError
+		var parsedFirstFrame bool
+		validatingReader := bytes.NewReader(packet.data)
+		for {
+			frame, err := s.frameParser.ParseNext(validatingReader, packet.encryptionLevel)
+			if err != nil {
+				quicErr, ok := err.(*qerr.QuicError)
+				if ok && quicErr.IsIgnorable() {
+					ignorableError = quicErr
+					break
+				}
+				return err
+			}
+			if frame == nil {
+				if !parsedFirstFrame && !updateSrcConnID {
+					return qerr.ToIgnorableError(fmt.Errorf("Packet contains only padding frames."))
+				}
+				break
+			}
+			parsedFirstFrame = true
+			if err := s.validateFrame(frame, packet.packetNumber, packet.encryptionLevel); err != nil {
+				quicErr, ok := err.(*qerr.QuicError)
+				if ok {
+					if quicErr.IsIgnorable() {
+						ignorableError = quicErr
+						break
+					}
+					if quicErr.IsRecoverable() {
+						recoverable = true
```

Comment:
Does this logic actually work?  It looks like a packet that generates a recoverable error is still fully processed through `s.handleFrame`, which could commit irrevocable changes to the session.